cdc/processor/pipeline/sink.go

```diff
@@ -142,7 +142,71 @@ func (n *sinkNode) flushSink(ctx pipeline.NodeContext, resolvedTs model.Ts) (err
 }
 
 func (n *sinkNode) emitEvent(ctx pipeline.NodeContext, event *model.PolymorphicEvent) error {
-	n.eventBuffer = append(n.eventBuffer, event)
+	if event == nil || event.Row == nil {
+		return nil
+	}
+
+	colLen := len(event.Row.Columns)
+	preColLen := len(event.Row.PreColumns)
+	config := ctx.ChangefeedVars().Info.Config
+
+	// This indicates that it is an update event,
+	// and after enable old value internally by default(but disable in the configuration).
+	// We need to handle the update event to be compatible with the old format.
+	if !config.EnableOldValue && colLen != 0 && preColLen != 0 && colLen == preColLen {
+		handleKeyCount := 0
+		equivalentHandleKeyCount := 0
+		for i := range event.Row.Columns {
+			if event.Row.Columns[i].Flag.IsHandleKey() && event.Row.PreColumns[i].Flag.IsHandleKey() {
+				handleKeyCount++
+				colValueString := model.ColumnValueString(event.Row.Columns[i].Value)
+				preColValueString := model.ColumnValueString(event.Row.PreColumns[i].Value)
```

Comment:
Just curious, no need to address, if we don't convert `Column.Value` to string, is there other (faster) way we can compare them.

Reply:
Interesting question! I think there are some complications with direct comparisons, because most values can be compared directly with ==, but for bytes values it can't be compared directly. So we still need a type assertion. But maybe a direct comparison would be faster than strings!